File: src/RESR.Models/Resources/Event.cs
namespace RESR.Models.Resources;

public sealed class Event : Resource
{
    public Event()
    {
        Type = ResourceType.Event;
    }

    public int IdEvent { get; set; }
    public string? Subtitle { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string? Address { get; set; }
    public int? IdDepartment { get; set; }
}

File: src/RESR.Frontend/src/shared/types/event.ts
import type { Category, PaginatedResponse, Resource, ResourceVisibility } from "@/shared/types/article";
import type { Department } from "@/shared/types/user";

export interface Event extends Resource {
  idEvent: number;
  visibility: ResourceVisibility;
  subtitle: string | null;
  startDate: string;
  endDate: string | null;
  address: string | null;
  department: Department | null;
}

export type EventPaginatedResponse = PaginatedResponse<Event>;
export type EventCategory = Category;
